Qgis server tutorial

tudorbarascu opened the pull request:
Hi, this is the core part of the QGIS Server tutorial.

I plan to also add QGIS Web Client howto  + using mapproxy in conjuction with QGIS Server and some other small ideas.

You can see the built documentation [here](http://docs.qtibia.ro/en/docs/training_manual/qgis_server/index.html).

The pull is opened so that I can get feedback and corrections.
Any suggestions/ideas are welcomed.

DelazJ commented:
@tudorbarascu if it's a tutorial, wouldn't the QGIS Training manual be a better place?

tudorbarascu commented:
@DelazJ I guess so. Will rework it to fit the training manual.

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+  The greatest advantage installing from packages is easyness. Another derived advantage is
+  that is far easier to keep deployment consistency. By having the same GNU/Linux flavor
+  pointing to the same software repositories users have obvious advantages: they can
+  share workflows, narrow down problems etc. far easier than otherwise).
```
No opening for the parenthesis

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+In our case, the easiest way to install the dependencies is to get the dependencies
+automatically from the qgis.org repo.
+In order to do so, you can follow the :ref:`qgis-server-repo-install`
+section untill you refresh the packages list.
```
Until

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+    remotes/origin/release-2_16
+  ...
+
+ You can switch do your desired QGIS version by doing ``git checkout release-2_16``
```
Switch to

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
````diff
@@ -0,0 +1,633 @@
+```
+sudo a2ensite qgis-ltr.demo.conf
+sudo systemctl restart apache2.service
+```
````
Not sure about the rendering you are looking for...

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+ <ServiceException code="Service configuration error">Service unknown or unsupported</ServiceException>
+ </ServiceExceptionReport>
+
+Now download the `demo data <https://github.com/tudorbarascu/qgis-server-tutorial-data/archive/master.zip>`_
```
Could this data be added to the [QGIS-Training-Data repo](https://github.com/qgis/QGIS-Training-Data)? Coherent if this chapter is moved to training doc.

tudorbarascu replied:
Will do, I'm currently still modifying the qgs project as I'm advancing with the tutorial so I'm keeping down the noise by working in another repo.

DelazJ replied:
Ok. Better add it once all is fixed indeed.

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+ </ServiceExceptionReport>
+
+Now download the `demo data <https://github.com/tudorbarascu/qgis-server-tutorial-data/archive/master.zip>`_
+and unzip the files to a directory of your choosing, like :file:`/home/web/projects`.
```
Choice?

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+
+ Pasting them into Mozilla Firefox works properly but other web browsers
+ like Chrome may add unwanted spaces between the ``field:parameter`` pairs.
+ So, if you encounter this issue you need can either use Firefox or modify
```
~need~

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+
+ This tutorial presents the URLs so that the audience can easily distinguish
+ the parameters and parameters values. While the normal format is
+ ``...&field1=value1&field2=value2&field3=value3`` this tutorial uses::
```
To ease reading (too much red thingies) and comparison between the two spelling, it could be nice to have this part of the url put inside a code-block (as below) instead of double back quotes.

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+As QGIS is also a :ref:`ogc-wms` you can create a new WMS server connection with the
+help of the above GetCapabilities url. See the :ref:`ogc-wms-servers` section on how to do it.
+
+By adding the ``countries`` WMS layer with the to your QGIS project you should get an image
```
~~with the~~

DelazJ replied:
Where does the "countries" layer come from? From your sample data I guess. Given it's the first time we are going to use those data, could be helpful to remind it I think

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+help of the above GetCapabilities url. See the :ref:`ogc-wms-servers` section on how to do it.
+
+By adding the ``countries`` WMS layer with the to your QGIS project you should get an image
+like the one bellow:
```
below

DelazJ commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,633 @@
+and we used the **REDLINING** feature to overlay a rectangle with the
+**QGIS Tutorial** label.
+
+TODO give credits at the end to alessandro, tim etc.
```
To be commented :)

DelazJ commented:
Added few comments but as you know this is not a side of qgis I know. My comments are then about typos and few newbies thoughts. I let experts comment the real content.

Any chance to have something for Windows users? I remember a [recent pull request](https://github.com/qgis/QGIS-Documentation/pull/1312)  aiming to address something like that and has been rejected. Maybe fixing what sounds wrong in the PR (I understood 64 bit was not a compatible target. What about 32?) and add it to the tutorial. Unless QGIS Server is incompatible with Windows (and in this case, it might be mentioned at the beginning of this chapter), it looks weird for a Windows user that follows the Training manual to have a whole chapter dedicated to \*nix environment without any disclaimer...

Thanks for tackling this, whatever

yjacolin commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,639 @@
+ branch. By default you're on the ``master`` branch. You can see at anytime what
+ branch you're on with the ``git branch`` command.
+
+ You can do a ``git branch -a`` to see all available branches (local and remote)::
```
I think this is too much details for a tutorial.

DelazJ replied:
Isn't a tutorial supposed to be detailed (unlike a manual :) )?

yjacolin replied:
Yes, but kept focus on the topic, giving more details on git branch concept is out of topic. 

Take care also that user should have git installed in his/her system. If you don't tell it, they can have issue here with the git clone.

tudorbarascu replied:
I wrote at the beginning of the build from source part that "This section of the tutorial is OPTIONAL and you can skip it by going to". It's for the more advanced users, who want to build from source. If they build from source, they should know about branches.

yjacolin commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,639 @@
+You can check where it has been installed by running ``locate qgis_mapserv.fcgi`` or
+``find / -name 'qgis_mapserv.fcgi'``.
+
+You should see these results::
```
*should* ? or `you may see these results` or something else? If people don't see such result they could think that there is a problem.

tudorbarascu replied:
locate uses a prebuilt database for searching. Thus, it is much faster than running find on the whole filesystem. However, if that database is not updated, it won't find anything so the users would need to do a ``sudo updatedb``. The database will probably be updated (this why I wrote should). However, in case the user doesn't see that result, he can try with the ``find`` command and he will surely find what he looks for.
I haven't given this details as they're out of scope and too cumbersome.
IMHO it should rest this way. If you still think I should add/remove something please say so.

yjacolin commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,639 @@
+
+   DocumentRoot /var/www/html
+
+   # Apache logs (different then QGIS Server log)
```
then -> than or from?

yjacolin commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,639 @@
+.. note::
+
+ If you installed QGIS Server without running an X Server (Linux Desktop part) and if you also
+ want to use the ``GetPrint`` command then you should install a fake X Server and tell Qgis
```
QGIS not Qgis :)

yjacolin commented on source/docs/user_manual/working_with_ogc/ogc_server_tutorial.rst:
```diff
@@ -0,0 +1,639 @@
+ sudo systemctl restart apache2.service
+
+We also need to tell the linux system that when a request is coming in for ``qgisplatform.demo``
+or for ``qgis-ltr.demo`` he should point to itself. We do that by adding in the `hosts
```
he -> it

yjacolin commented:
@tudorbarascu really nice tutorial. As @DelazJ said before, it is not really fit for documentation, but imho it is not fit also for trainings: the structure and contents are different from all other chapter.

I propose to add it as an appendix of the user-manual as we can keep reference uptodate with the manuel content.

DelazJ commented:
@yjacolin : 
> I propose to add it as an appendix of the user-manual as we can keep reference uptodate with the manuel content.

Sorry but I do not share that. Put it in the [appendix](http://docs.qgis.org/testing/en/docs/user_manual/appendices/appendices.html)  (?) and I'm afraid it'll be hidden (even with links to it). This word doesn't have any meaning for me and I'm not sure it's the place people expects such doc.  
And @tudorbarascu answered above (on moving to Training manual):
> @DelazJ I guess so. Will rework it to fit the training manual.

I suppose once the content is fixed, it'll be "simply" a matter of sectioning (and adding few comments to link lessons) to fit Training manual structure. Note that Training manual has nothing on QGIS Server yet. And I suppose user may need that.

yjacolin commented:
```An appendix is a section at the end of a book that includes supplementary information that the author or publisher thinks may be of interest to the reader, but is either too tangential or too detailed to be worked into the actual text```

Tutorials fits exactly this definition.

As a training manager, I don't think that a tutorials can be changed to a training just adding few sections. :) This is really a different concept, structure and much more work!

I agree that training manual needs a section on QGIS Server, but it is better to learn how to configure it, use it than build it from source. If you want to add it in training manual, you should remove the "build from source" section, split it in several parts, add goals, exercices and results. Much more work, if @tudorbarascu is ok to work on this way, he is free to do so.

DelazJ commented:
Hummmm! I get your points. And you may be right on the differences you exposed and the amount of work needed. Up to Tudor to see what option he takes.

I'm still not convinced by the Appendix chapter as target (but this is not a real concern). I quickly read again the PR (note that I'm not QGIS Server user) and I wonder if there's not duplicates (or a lot of overlaps) between the PR and current [QGIS-Server doc](http://docs.qtibia.ro/en/docs/user_manual/working_with_ogc/ogc_server_support.html#qgis-as-ogc-data-server). Though I know @tudorbarascu has recently worked on this chapter, aren't there possibilities to insert missing/valuable contents from the tutorial into the existing chapter (there are already sections for installation, configuration, parameters... that could be better reorganized in the TOC or expanded)? 
**NOTA**: I'm not saying that the tutorial is useless as is, Tudor.

tudorbarascu commented:
@DelazJ I know they are some overlaps and I actually thought to rework or discontinue [this part](http://docs.qgis.org/testing/en/docs/user_manual/working_with_ogc/ogc_server_support.html#qgis-server-installation-on-debian-ubuntu) after everything is complete.

I actually was thinking of reworking it to fit the training manual (with the **OPTIONAL** build from source part) and put it right after the **Module: Online Resources**.

I could go the appendix way as it would be easier for me. Also note that it's still not finished, as I plan on doing the QGIS Web Client part, WFS, mapproxy. However, it would be really nice to settle on the structure and where it should go as I can rework the structure now but later on it would take too much time.

DelazJ commented:
@tudorbarascu, sorry for the very late reply.
> I actually was thinking of reworking it to fit the training manual (with the OPTIONAL build from source part) and put it right after the Module: Online Resources.

I'm ok with that. 👍 
>  could go the appendix way as it would be easier for me. Also note that it's still not finished, as I plan on doing the QGIS Web Client part, WFS, mapproxy.

Huge and nice program! Thanks.

tudorbarascu commented:
I've moved the tutorial from the user manual to the training manual.
I'm trying to fit it in as good as possible but I think there's also room for improvement.
I am planning to extend it but I first want to have the basic module merge-able.
I also deleted the build from source part.

You can check out how it looks now at http://docs.qtibia.ro/en/docs/training_manual/qgis_server/index.html

tudorbarascu commented:
@DelazJ @yjacolin I think it's more or less good enough to merge so you can check it out again.
I want to add other chapters incrementally, after a successful merge as I don't have the time now.
I plan to also backport it to the 2.18 branch + updating it in the future to synchronize with the changes in master.
Thoughts?

yjacolin commented on source/docs/training_manual/qgis_server/index.rst:
```diff
@@ -0,0 +1,16 @@
+
+In this module, we'll cover how to install and use QGIS Server.
+
+For a introduction to what QGIS Server is (see the :ref:`label_qgisserver` section)
```
aN introduction

yjacolin commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,316 @@
+|LS| Serving WMS
+===============================================================================
+
+Let's download the `demo data <https://github.com/tudorbarascu/qgis-server-tutorial-data/archive/master.zip>`_
```
Data should be in QGIS repository, could you merge it to https://github.com/qgis/QGIS-Training-Data

yjacolin commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,316 @@
+
+.. warning::
+
+ This module presents the URLs so that the audience can easily distinguish
```
I will add this as a note before the texte: ``Let's make a WMS GetCapabilities request in the web browse [..]``

yjacolin commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,316 @@
+ Pasting them into Mozilla Firefox works properly but other web browsers
+ like Chrome may add unwanted spaces between the ``field:parameter`` pairs.
+ So, if you encounter this issue you can either use Firefox or modify
+ the URLs so that they're in a one line format.
```
`in an one line format.` or `in one line format.`

yjacolin commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,316 @@
+
+The log files are simply text files so you can use a text editor to check them
+out. You can also use the `tail` command in a terminal like so::
+
```
could you use code-block instead ::? see lexer list here http://pygments.org/docs/lexers/. In that cas you canr use 
```.. code-block:: shell```

yjacolin commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,316 @@
+
+ * If you look at the logs in the following sections you should get a better
+   understanding on what's happening.
+ * It would be interresting for you to restart Apache and look in the QGIS
```
* interresting?
* restart is not useful, graceful is ok

yjacolin commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,316 @@
+
+* Based on the request above, let's replace the ``countries`` layer with another.
+* In order to see what other layers are available you could open up the
+  :file:`world.qgs` project in QGIS and see. Keep in mind though that the WMS
```
`and see` ->  `and look at its contents`?

yjacolin commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,316 @@
+GetPrint requests
+-----------------
+
+One very nice feature of QGIS Server is that it make use of the QGIS Desktop
```
it makeS

yjacolin commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,316 @@
+
+* As user ``qgis`` go to the home directory with ``cd /home/qgis``.
+* Download the QWC project with ``git clone git://github.com/qgis/QGIS-Web-Client.git``.
+  If you don't have ``git`` installed you can install it with ``sudo apt-get install -y git``.
```
Why just ask to download an archive tarball?

tudorbarascu replied:
I used git so that I can make use of the ``git diff`` command further down the road, when doing some configuration changes on the QWC.

yjacolin commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+  run multiple versions of QGIS (consequently QGIS Server) at the same time.
+  Also, you can build QGIS on top of other components versions (e.g. newer
+  GDAL, GEOS etc.) which could get you new features or help you narrow down
+  bugs. Furthermore, if you're a developper and you want to do some changes
```
is it really worth to explain this in a training manual and in QGIS server chapter?

tudorbarascu replied:
Not necessarily. On the other hand, this part it's only written in mailing lists, stackexchange etc. and I think it has to pop somewhere in the documentation as it was something that really bugged me at first. In time, I would also like to talk about docker etc.
Of course, we can just remove it but I think we also have to consider the more advanced users.

yjacolin commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+
+First add the QGIS repository by creating the
+:file:`/etc/apt/sources.list.d/debian-qgis.list` file with the following
+content::
```
can we focus on graphical user interface? What about Windows user?

tudorbarascu replied:
Don't understand what's needed here.

DelazJ replied:
I suppose @yjacolin refers to the possibility to use graphical application (eg, Synaptic?) to set the package list
About Windows user, the introduction already states that linux will be the target so, though i'd expect Windows user to not be left aside, not mention them keeps coherence

tudorbarascu replied:
Oh! Although the tutorial installs the Desktop part for commodity sake (so that the participants don't have to use a second machine with Desktop) this is a server tutorial and I want to keep things in command line as much as possible.
I added a comment that in production is not recommended to use the Desktop on the same machine as the Server.
What do you think?

yjacolin commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+ :ref:`server_env_variables` section.
+
+Let's now create the directories that will store the QGIS Server logs and
+the authentication database::
```
could you replace here and ther `::` by `..code-block:: [shell | ...]`?

tudorbarascu replied:
I don't understand exactly what you want here ;). Thanks for reviewing.

tudorbarascu replied:
@DelazJ @yjacolin I can replace ``::`` with ``..code-block:: shell``, of course.

What's the reason that we want it this way?
Thanks

DelazJ replied:
I don't know how it works for Romanian or other languages but I saw many French translation that didn't follow the use of double colon at the end, hence the code block is not rendered with its formatting. Reason why I opted to hide this punctuation in translatable strings.
Also, using this syntax makes the commands and comments colored according to the language used, instead of a global gray text

tudorbarascu replied:
@DelazJ Makes sense. I did the changes. Thanks for the info.

yjacolin commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+``example.com``, etc.) but for simplicity sake we're going to use a letter from
+the alphabet, let's say ``x``.
+
+* Let's first point the ``x`` name to answer to the localhost IP. We can do that
```
Does `x` is really a good name? `QGIS` is not better?

tudorbarascu replied:
I wanted to exemplify any name would do and ``x`` seemed ok.
I already have ``qgisplatform.demo`` so writing ``qgis`` is still too alike and I'm unsure that the user will get the point. Could do with another name if you have a better suggestion or we could use qgis as you suggested. @DelazJ Thoughts?

tudorbarascu replied:
@DelazJ You missed reviewing here ;)

DelazJ replied:
<story>actually I fall asleep before I finished the review and just pushed what I've written so far this morning when I woke up hence, I haven't review beyond my last comment ( execute qgis server)<\story> will try to finish till this evening

DelazJ replied:
I agree that `qgis` is too alike for the reader but x might also be misleading due to x server mentioned in a previous note. A more-than-one-letter name could be better (foo, test, demo, bar, myhost...).

yjacolin commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+|moderate| |TY| Create another virtual host
+-------------------------------------------------------------------------------
+
+Let's create another Apache virtual host pointing to QGIS Server. You can
```
Why do you want to create another Apache vhost?

tudorbarascu replied:
To ensure the user understand what virtual hosts are, what the hosts file does etc.

yjacolin commented:
Here are my comments. Some are just small comment, other need feedback (two vhosts, linux focus).

This is a great chapter, thank you very much!

tudorbarascu commented:
@yjacolin Thanks a lot for reviewing. I've addressed the majority of the comments, some I didn't get and for others I'm unsure and I'm waiting for a second feedback.

tudorbarascu commented:
Hi @DelazJ Can you take a look at this?

DelazJ commented:
Hi @tudorbarascu sorry I forgot u. Will do later. Is your website still up to date with the PR? A build will surely help to see the big picture of the PR. Thanks

tudorbarascu commented:
@DelazJ Thanks for looking into it.
The website is up to date!

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+
+ One way of creating the above file is to run a terminal, then type in it
+ ``sudo gedit /etc/apt/sources.list.d/debian-qgis.list``, which opens the
+ gedit text editor. Add the content in the editor window and save.
```
Is this note needed? I'm afraid it overloads the description. I suppose any *nix user should know how to add package sources

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+
+Run these commands to add the qgis.org repository public key to your apt keyring::
+
+ sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-key 073D307A618E5811
```
warning, it seems that our key server is no longer valid (see #1790)

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+
+You can now run the ``sudo apt-get update`` command to refresh the packages list and
+``sudo apt-get dist-upgrade`` to upgrade the packages.
+
```
Actually, all this section is somehow redundant with QGIS website download page you mentioned above. Couldn't it be summarized by just sending people to that page? And if still needed, simply keep the couple of important commands

tudorbarascu replied:
I can just send people to that page, of course. However, I had noticed beginners are baffled by all the options and sometimes have a hard time finding the proper combinations. That's why I thought on presenting how it should be done on one system only.
@DelazJ @yjacolin Your call: Make this section a link to the http://qgis.org/en/site/forusers/alldownloads.html#debian-ubuntu or generally leave it like it is.

DelazJ replied:
My feeling is that if QGIS Team should provide install instructions from package, it might be done at a single place, the clearer and most complete possible. This ensures coherence and eases updates and fixes. If there are things beginners do not understand let's fix it there; there are more users who likely will install qgis (or QGIS server) from the website instructions than from within the training manual (of QGIS Server) so I'd say: 

1. fix http://qgis.org/en/site/forusers/alldownloads.html#debian-ubuntu with missing bits (how do I install/upgrade/downgrade, anything you feel is unclear or not needed...) 
2. just point to that cleaned QGIS website and remove instructions here
3. maybe simply keep here the commands to install qgis server (for the recall)

tudorbarascu replied:
@DelazJ @yjacolin I reduced the install part to a minimum. I will update the website instructions later (probably at the Essen hackfest).

Thanks for reviewing.

DelazJ replied:
Thanks @tudorbarascu 
I think the keyserver code block part should also be removed as this key seems to already be out of date (according to https://github.com/qgis/QGIS-Website/issues/374 and https://github.com/qgis/QGIS-Documentation/issues/1790) or will be outdated in august (according to http://qgis.org/fr/site/forusers/alldownloads.html#debian-ubuntu)

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+
+   The ``-y`` parameter means that you accept from command line the install of all
+   the enumerated packages, bypassing the prompt you usually get.
+
```
Again, I wonder if this should be in the scope of the training. It looks like the aim is to teach people setting package list and not QGIS Server stuffs. People actually do not want to play with versions; if I follow this learning, it'd be to setup a working environment so i'll likely skip this section. see http://docs.qgis.org/testing/en/docs/training_manual/spatial_databases/spatial_functions.html
Do not misinterpret me, these are knowledge I'd have liked to see the first time I tried to install different version (especially downgrade my installation) on my Ubuntu --> might be invaluable in qgis download instructions page. Currently, there's no real statement explaining that you can't have different versions in linux and how to move from one to other so this text placed there will help many people.

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+should output something like ``/usr/lib/cgi-bin/qgis_mapserv.fcgi``.
+
+Optionally, if you want to do a command line test at this time you can run
+``/usr/lib/cgi-bin/qgis_mapserv.fcgi`` which should output something like::
```
how do I run /usr/lib/cgi-bin/qgis_mapserv.fcgi? What's the command? or is it a double-click?

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,417 @@
+  <ServiceException code="Service configuration error">Service unknown or unsupported</ServiceException>
+ </ServiceExceptionReport>
+
+This is a good thing, it tells you we're on the right track.
```
In case of failure, it could be nice to know why or how (links) to debug

tudorbarascu replied:
I redid this part based on your comment.

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,322 @@
+
+.. note::
+
+ See what some of the configuration options in the Server
```
see ~~what~~ some?

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,322 @@
+  logs go as otherwise the logs for the two virtual hosts would be shared but this is
+  optional.
+* Let's now enable the virtual host with ``sudo apt-get a2ensite x.conf`` and the
+  reloading the Apache service with ``sudo systemctl reload apache2``.
```
and ~~the~~ reload

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,322 @@
+* Let's now enable the virtual host with ``sudo apt-get a2ensite x.conf`` and the
+  reloading the Apache service with ``sudo systemctl reload apache2``.
+* If you try again to access the http://x/cgi-bin/qgis_mapserv.fcgi url you'll
+  notice everything is working now!.
```
colon to delete

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,330 @@
+ sets the default project to :file:`/home/qgis/projects/world.qgs`. However, we
+ made use of the **map** parameter to be explicit and to show it can be used to
+ point at any project. In the above request, QGIS Server outputs the same
+ response if you don't use the **map** parameter.
```
Couldn't find these information (the use of map) in the previous lesson.

tudorbarascu replied:
In the previous lesson there's only the config. The map parameter is in the request above. I've reworked this part to be more explicit.

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,330 @@
+ &REQUEST=GetCapabilities
+ &map=/home/qgis/projects/world.qgs
+
+.. note::
```
Should this be a note?

tudorbarascu commented:
@DelazJ I've made the requested changes. Could you please take a peep? Thanks!

DelazJ commented on source/docs/training_manual/qgis_server/install.rst:
```diff
@@ -0,0 +1,322 @@
+Apache is now configured.
+
+|moderate| |TY| Create another virtual host
+-------------------------------------------------------------------------------
```
|TY| or |FA|?

tudorbarascu replied:
I think TY is better as it fixes the knowledge already written, without introducing something new. Am I getting this wrong?

DelazJ replied:
From What I've seen, "Try Yourself" is a kind of exercise section (e.g. http://docs.qtibia.ro/en/docs/training_manual/introduction/overview.html#basic-ty-1) with a link to correction. This is not the case in this section and more globally, in this chapter (which somehow reminds me the _"reluctance"_ of @yjacolin to add this as a training material).

tudorbarascu replied:
So, should I make it FA?

DelazJ replied:
Actually I finally found some cases where there was no exercise...

tudorbarascu replied:
The actual "checking" process is embedded as the participant can test with ping (for the host file) + curl or a link (for the apache conf) file.

DelazJ replied:
> So, should I make it FA?

I'd say yes, it'll be more consistent because we don't let the reader do it by himself, we still show him step by step what to do, so "Follow along"

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,330 @@
+
+.. only:: html
+
+  **Figure countries WMS layer from QGIS Desktop:**
```
We no longer used a html only top title for figures. I don't recall very well all the reasons but it might have to do with the aim to upgrade sphinx and use a single numbering system beside the image caption (common to html and pdf).
Instead, you could add an anchor (we do for images but it's not a requirement, just for consistency).

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,330 @@
+ QGIS Server serves layers that are defined in the :file:`world.qgs` project.
+ By opening the project with QGIS you can see there are multiple styles for
+ the countries layer. QGIS Server is also aware of this and you can choose
+ the style you want in your request.  The `classified_by_population` style
```
two spaces after the colon
better double the *`* character if you want your text with a special rendering

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,330 @@
+-------
+
+When you're setting up a server, the logs are always important as they show
+you what's going on. We have setup in the ``*.conf`` file the following logs:
```
:file:`*.conf`

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,330 @@
+
+A simple request looks like:
+
+.. code-block:: guess
```
I didn't know about this value. How does it work (it seems to render some highlighting in html)?

tudorbarascu replied:
The guess parameter lets Pygments guess the lexer based on contents, only works with certain well-recognizable languages.
See http://www.sphinx-doc.org/en/1.5.1/markup/code.html 
I was in doubt about the possibilities so I tried guess after reading about it and after checking the build, I liked what I saw.

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,330 @@
+
+|moderate| |TY| Change the Image and Layers parameters
+-------------------------------------------------------------------------------
+
```
Any reason the text below is in a list?

tudorbarascu replied:
I noticed that other TY parts were usually like this and I wanted to be consistent. I changed it now.

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -0,0 +1,330 @@
+  in the ``GetCapabilities`` XML response.
+* One of the layer names that you could find and works is ``countries_shapeburst``.
+  You may find others but keep in mind some may not be visible at such a small
+  scall so you yould get a blank image as response.
```
scale
could

DelazJ commented:
Sorry @tudorbarascu I didn't finish the review (I should have let u know) but now I think I'm done!!! If ever there are things to improve, I think this could be done in a second step...

tudorbarascu commented:
@DelazJ No worries. Thanks again for your valuable input. I did the latest changes based on your observations.

DelazJ commented on source/docs/training_manual/qgis_server/wms.rst:
```diff
@@ -61,9 +61,7 @@ or the  :ref:`ogc-wms-servers` section on how to do it.
-.. only:: html
-
-  **Figure countries WMS layer from QGIS Desktop:**
+**Figure countries WMS layer from QGIS Desktop:**
```
not even the title is kept, otherwise it shows up in pdf which is imho uglier.
see https://github.com/qgis/QGIS-Documentation/pull/1533/commits/5f7806b44b83a69afa05ac7f045159646e0b2741

tudorbarascu commented:
Done.
